app/ota_client.py

```diff
@@ -206,25 +205,58 @@ class OtaClientUpdatePhase(Enum):
 
 class OtaClientStatistics(object):
     def __init__(self):
-        self.clear()
+        self._lock = Lock()
+        self._slot = self._init_statistics_storage()
 
-    def clear(self):
-        self.total_files = 0
-        self.files_processed = 0
+    @staticmethod
+    def _init_statistics_storage() -> dict:
+        return {
+            "total_files": 0,
+            "files_processed": 0,
+            "files_processed_copy": 0,
+            "files_processed_link": 0,
+            "files_processed_download": 0,
+            "file_size_processed_copy": 0,
+            "file_size_processed_link": 0,
+            "file_size_processed_download": 0,
+            "elapsed_time_copy": 0,  # in milliseconds
+            "elapsed_time_link": 0,  # in milliseconds
+            "elapsed_time_download": 0,  # in milliseconds
+            "errors_download": 0,
+        }
 
-        self.files_processed_copy = 0
-        self.files_processed_link = 0
-        self.files_processed_download = 0
+    def get_snapshot(self):
+        """
+        return a copy of statistics storage
+        """
+        return self._slot.copy()
 
-        self.file_size_processed_copy = 0
-        self.file_size_processed_link = 0
-        self.file_size_processed_download = 0
+    def set(self, attr: str, value):
+        """
+        set a single attr in the slot
+        """
+        with self._lock:
+            self._slot[attr] = value
```

Comment:
I didn't know until recently, but this operation doesn't require a lock!
https://docs.python.org/3/faq/library.html#what-kinds-of-global-value-mutation-are-thread-safe